src/OpenLoco/Economy.cpp

```diff
@@ -1,21 +1,101 @@
 #include "Economy.h"
 #include "Interop/Interop.hpp"
+#include "Ui/WindowManager.h"
+#include "Ui/WindowType.h"
 
 using namespace OpenLoco::Interop;
 
 namespace OpenLoco::Economy
 {
+    // Inflation rates?
+    static const uint32_t _4FDEC0[32] = {
+        20,
+        20,
+        20,
+        20,
+        23,
+        20,
+        23,
+        23,
+        20,
+        17,
+        17,
+        20,
+        20,
+        20,
+        20,
+        20,
+        20,
+        20,
+        20,
+        20,
+        20,
+        20,
+        20,
+        20,
+        20,
+        20,
+        20,
+        20,
+        20,
+        20,
+        20,
+        20,
+    };
+
+    static loco_global<uint32_t[32], 0x00525E5E> currencyMultiplicationFactor;
+
+    // NB: _525EDE[2] is dword_525E66, referred to by four subs for (price) computation.
+    static loco_global<uint32_t[32], 0x00525EDE> _525EDE;
+
     // 0x0046E239
+    // NB: called in sub_46E2C0 below, as well in openloco::date_tick.
     void sub_46E239()
     {
-        call(0x0046E239);
+        for (uint8_t i = 0; i < 32; i++)
+        {
+            uint32_t ebx = currencyMultiplicationFactor[i];
+
+            // Inflating prices?
+            uint64_t edx_eax = _525EDE[i] * _4FDEC0[i];
+            edx_eax >>= 12;
+
+            // Storing two halves of the money struct?
+            _525EDE[i] += static_cast<uint32_t>(edx_eax);
+            currencyMultiplicationFactor[i] += static_cast<uint32_t>(edx_eax >> 32);
```

Comment:
this isn't quite correct. There is a carry from 525ede.
You'll need to do:
```
uint64_t new525 = _525EDE[i] + static_cast<uint32_t>(edx_eax);
_525EDE[i] = static_cast<uint32_t>(new525);
currencyMultiplicationFactor[i] += static_cast<uint32_t>(edx_eax >> 32) + (new525 >> 32);
```

Reply:
Wait, there is a much easier way of doing this its just simple maths.

```
uint64_t existingInflation = _525EDE[i] + (currencyMultiplicationFactor[i] << 32);
uint64_t additonalInflation = (existingInflation *  _4FDEC0[i]) / 4096;
uint64_t newInflation = existingInflation + additionalInflation;
_525EDE[i] = static_cast<uint32_t>(newInflation);
currencyMultiplicationFactor[i] = static_cast<uint32_t>(newInflation>> 32);
```

Reply:
Oh wait but CS didn't do his calculation correctly. So this is wrong.

Reply:
```
        for (uint8_t i = 0; i < 32; i++)
        {
            currencyMultiplicationFactor[i] += (static_cast<uint64_t>(_4FDEC0[i]) * currencyMultiplicationFactor[i]) >> 12;
        }
```

Reply:
Wow. I'm pleasantly surprised that chunk of code just boiled down to a oneliner in the end. Thank you very much!